sql/moz-fx-data-shared-prod/accounts_db_derived/accounts_aggregates_v1/query.sql

```diff
@@ -0,0 +1,65 @@
+-- Source tables of this query aren't partitioned and are overwritten daily.
+-- To make this query idempotent we need to make sure that the data for the
+-- current submission_date is not overwritten. If data for the current
+-- submission_date is already present in the table, this query returns the
+-- existing data. Otherwise, it returns the new data aggregated from the
+-- source tables.
+WITH current_data AS (
+  SELECT
+    submission_date,
+    active,
+    two_factor_auth_enabled,
+    recovery_keys_enabled,
+    total_accounts,
+  FROM
+    `moz-fx-data-shared-prod.accounts_db_derived.accounts_aggregates_v1`
+  WHERE
+    submission_date = @submission_date
+),
```

Comment:
Why is the current data necessary here if only to UNION since the table is already partitioned by `submission_date`? Is the query meant to run multiple times per-day?

Reply:
Query is meant to be run once a day. I wanted to make sure it won't overwrite past partitions if the task gets cleared in Airflow. Since source tables are not partitioned, re-runs could overwrite history with today's aggregate.
To ensure this doesn't happen I need to check if the partition we're trying to write to already has any data - hence the current data CTE.

Reply:
Hmm it seems like the table doesn't really need to be partitioned then? With a `last_updated` column and a check to see if it ever changes by more than a day or two past `CURRENT_DATE` (in which case we could use BQ time travel)?